test/src/missle_launcher_test.cpp

```diff
@@ -5,22 +5,68 @@
 #include "mocks/mock_missile_launcher_listener.h"
 
 using ::testing::_;
+using ::testing::Property;
+using ::testing::InSequence;
 
-class MissleLauncherTest : public ::testing::Test {
+class MissileLauncherTest : public ::testing::Test {
 protected:
+    MissileLauncher* launcher;
+    MockMissileLauncherListener* launcher_listener;
+
     virtual void SetUp() {
+        launcher = new MissileLauncher() ;
+        launcher_listener = new MockMissileLauncherListener();
+        launcher->add_listener(launcher_listener);
     }
 
     virtual void TearDown() {
+        delete launcher_listener;
     }
 };
 
-TEST(MissileLauncherTest, check_listeners_notified_onlaunch) {
-    MockMissileLauncherListener launcher_listener;
-    EXPECT_CALL(launcher_listener, notify_missile_launched(_))
-        .Times(1);
+TEST_F(MissileLauncherTest, multiple_listeners_notified) {
+    MockMissileLauncherListener launcher_listener2;
+
+    EXPECT_CALL(*launcher_listener, notify_missile_launched(_));
```

Comment:
To clarify, these 2 EXPECT_CALL checks are expected to fail right? Because the notify_missile_launched will only get called when a missile is created (which happens after these 2 EXPECT_CALL checks).